fulp_modules/bloodsuckers/bloodsucker/bloodsucker_coffin.dm

```diff
@@ -0,0 +1,176 @@
+	name = "metal coffin"
+	desc = "A big metal sardine can inside of another big metal sardine can, in space."
+	icon_state = "metalcoffin"
+	icon = 'fulp_modules/bloodsuckers/icons/vamp_obj.dmi'
+	resistance_flags = FIRE_PROOF | LAVA_PROOF
+	open_sound = 'sound/effects/pressureplate.ogg'
+	close_sound = 'sound/effects/pressureplate.ogg'
+	breakout_time = 300
+	pryLidTimer = 200
+	material_drop = /obj/item/stack/sheet/metal
+	material_drop_amount = 5
+	armor = list("melee" = 40, "bullet" = 15, "laser" = 50, "energy" = 0, "bomb" = 10, "bio" = 0, "rad" = 0, "fire" = 70, "acid" = 60)
+
+//////////////////////////////////////////////
+
+/obj/structure/closet/crate/proc/ClaimCoffin(mob/living/claimant) // NOTE: This can be any "closet" that you are resting AND inside of.
+	// Bloodsucker Claim
+	var/datum/antagonist/bloodsucker/bloodsuckerdatum = claimant.mind.has_antag_datum(/datum/antagonist/bloodsucker)
+	if(bloodsuckerdatum)
+		// Vamp Successfuly Claims Me?
+		if(bloodsuckerdatum.ClaimCoffin(src))
+			resident = claimant
+			anchored = 1					// No moving this
+
+/obj/structure/closet/crate/coffin/Destroy()
+	UnclaimCoffin()
+	return ..()
+
+/obj/structure/closet/crate/proc/UnclaimCoffin()
+	if (resident)
+		// Vamp Un-Claim
+		if (resident.mind)
+			var/datum/antagonist/bloodsucker/bloodsuckerdatum = resident.mind.has_antag_datum(/datum/antagonist/bloodsucker)
+			if (bloodsuckerdatum && bloodsuckerdatum.coffin == src)
+				bloodsuckerdatum.coffin = null
+				bloodsuckerdatum.lair = null
+			to_chat(resident, "<span class='danger'><span class='italics'>You sense that the link with your coffin, your sacred place of rest, has been broken! You will need to seek another.</span></span>")
+		resident = null // Remove resident. Because this object isnt removed from the game immediately (GC?) we need to give them a way to see they don't have a home anymore.
+
+/obj/structure/closet/crate/coffin/can_open(mob/living/user)
+	// You cannot lock in/out a coffin's owner. SORRY.
+	if (locked)
+		if(user == resident)
+			if (welded)
+				welded = FALSE
+				update_icon()
+			//to_chat(user, "<span class='notice'>You flip a secret latch and unlock [src].</span>") // Don't bother. We know it's unlocked.
+			locked = FALSE
+			return 1
+		else
+			playsound(get_turf(src), 'sound/machines/door_locked.ogg', 20, 1)
+			to_chat(user, "<span class='notice'>[src] is locked tight from the inside.</span>")
+	return ..()
+
+/obj/structure/closet/crate/coffin/close(mob/living/user)
+	if (!..())
+		return FALSE
+	// Only the User can put themself into Torpor. If already in it, you'll start to heal.
+	if ((user in src))
+		var/datum/antagonist/bloodsucker/bloodsuckerdatum = user.mind.has_antag_datum(/datum/antagonist/bloodsucker)
+		if (bloodsuckerdatum)
+			LockMe(user)
+			if (!bloodsuckerdatum.coffin && !resident) //Claim?
+				switch(alert(user,"Do you wish to claim this as your coffin? [get_area(src)] will be your lair.","Claim Lair","Yes", "No"))
+					if("Yes")
```

Comment:
what if they stall the input here? they could, _gasp_, anchor a locker remotely! the horror!